spec/client_spec.rb

```diff
@@ -41,6 +41,22 @@
     subject.call(action, params)
   end
 
+  it 'does not have an Authorization header when auth: is not specified' do
+    expect(subject.send :headers).to_not include('Authorization') 
+  end
+
+  context 'auth: is specified' do
+    let(:mock_key) { '00000000-0000-0000-0000-000000000000' }
+
+    before do
+      allow(subject).to receive(:auth).and_return(mock_key)
```

Comment:
I think this is the actual behavior you want to test against, so I wouldn't intercept it using `allow`.  I think what you want is something similar to `client_with_config` as seen above.  You want to ensure that you can pass in `auth` header so I think you want an example like

```ruby
context 'auth header' do
  let(:auth_key) { 'somekey' }
  let(:client_with_auth_key) { described_class.new(auth: auth_key) }

  it 'allows and exposes auth header configuration' do
    expect(client_with_auth_key.auth).to eq(auth_key)
  end
end
```

As far as calling the private `headers` method, I see the logic there, but you're not supposed to test internal implementation.  I see that I'm missing a test here as far as valid calls to `RestClient`, so if you can simplify your test according to my suggestion above then I can add a better test including url/headers against `RestClient` after this is merged in.